layerscape: upgrade kernel to 4.14

yangbolu1991 opened the pull request:
Sorry for late to send out kernel 4.14 upgrading patches.

Just copied commit message here to let you know what patches-4.14 included. Also attached all boards test logs except Traverse LS1043 Board since I didn't have that board.
[openwrt_upstreaming_4.14_log.tar.gz](https://github.com/openwrt/openwrt/files/2606467/openwrt_upstreaming_4.14_log.tar.gz)

Please help to review and merge. Thanks a lot.

    layerscape: upgrade kernel to 4.14

    This patch is to upgrade kernel to 4.14 for layerscape.
    patches-4.14 for layerscape included two categories.

    - NXP Layerscape SDK kernel-4.14 patches

      All patches on tag LSDK-18.09-V4.14 were ported to OpenWrt
      kernel. Since there were hundreds patches, we had to make
      an all-in-one patch for each IP/feature.
      See below links for LSDK kernel.
      https://lsdk.github.io/components.html
      https://source.codeaurora.org/external/qoriq/qoriq-components/linux

    - Non-LSDK kernel patches

      Other patches which were not in LSDK were just put in patches-4.14.
      Kept below patches from patches-4.9.
      303-dts-layerscape-add-traverse-ls1043.patch
      821-add-esdhc-vsel-to-ls1043.patch
      822-rgmii-fixed-link.patch

      Renamed and rebase them as below in patches-4.14,
      303-add-DTS-for-Traverse-LS1043-Boards.patch
      712-sdk-dpaa-rgmii-fixed-link.patch
      824-mmc-sdhci-of-esdhc-add-voltage-switch-support-for-ls.patch

biwenli committed:
layerscape: upgrade kernel to 4.14

This patch is to upgrade kernel to 4.14 for layerscape.
patches-4.14 for layerscape included two categories.

- NXP Layerscape SDK kernel-4.14 patches

  All patches on tag LSDK-18.09-V4.14 were ported to OpenWrt
  kernel. Since there were hundreds patches, we had to make
  an all-in-one patch for each IP/feature.
  See below links for LSDK kernel.
  https://lsdk.github.io/components.html
  https://source.codeaurora.org/external/qoriq/qoriq-components/linux

- Non-LSDK kernel patches

  Other patches which were not in LSDK were just put in patches-4.14.
  Kept below patches from patches-4.9.
  303-dts-layerscape-add-traverse-ls1043.patch
  821-add-esdhc-vsel-to-ls1043.patch
  822-rgmii-fixed-link.patch

  Renamed and rebase them as below in patches-4.14,
  303-add-DTS-for-Traverse-LS1043-Boards.patch
  712-sdk-dpaa-rgmii-fixed-link.patch
  824-mmc-sdhci-of-esdhc-add-voltage-switch-support-for-ls.patch

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>
Signed-off-by: Biwen Li <biwen.li@nxp.com>

biwenli committed:
layerscape: add LS1021AIOT board support

The LS1021A-IoT gateway reference design based on the
QorIQ LS1021A processor is a purpose-built, small
footprint hardware platform with a wide array of
high-speed connectivity and low-speed serial interfaces
to support secure delivery of IoT services for home,
business or other commercial location.

- Combines standards-based, open source software with a
  feature-rich IoT gateway design to establish a common,
  open framework for secured IoT service delivery and
  management.

- Provides a wide assortment of high-speed and serial-based
  connectivity in a compact, highly secure design.

- High efficiency through the use of the Arm-based QorIQ
  LS1021A embedded processor.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>
Signed-off-by: Biwen Li <biwen.li@nxp.com>

yangbolu1991 committed:
layerscape: use MKUBIFS_OPTS for per-device ubi parameters

UBIFS_OPTS couldn't be used for per-device ubi parameters.
Let's use MKUBIFS_OPTS instead.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>

hauke commented on target/linux/layerscape/armv7/config-4.14:
```diff
@@ -0,0 +1,925 @@
+CONFIG_CACHE_L2X0=y
+# CONFIG_CACHE_L2X0_PMU is not set
+# CONFIG_CFS_BANDWIDTH is not set
+CONFIG_CGROUPS=y
```
Please do not activate CGROUPS by default, there is the KERNEL_CGROUPS config option in OpenWrt ton activate it.

yangbolu1991 replied:
Will remove it.

hauke commented on target/linux/layerscape/armv7/config-4.14:
```diff
@@ -0,0 +1,925 @@
+CONFIG_GPIO_TPS6586X=y
+CONFIG_GPIO_TPS65910=y
+CONFIG_GPIO_TWL4030=y
+CONFIG_GPIO_XILINX=y
```
Do you have all these GPIO controllers on your boards? If not just deactivate these options.

yangbolu1991 replied:
layerscape didn't use them. Will remove all.

hauke commented on target/linux/layerscape/armv7/config-4.14:
```diff
@@ -0,0 +1,925 @@
+CONFIG_LS_SOC_DRIVERS=y
+CONFIG_LZO_COMPRESS=y
+CONFIG_LZO_DECOMPRESS=y
+CONFIG_MACVLAN=y
```
macvlan is normally activated by kmod-macvlan, no need to compile this into the kernel binary.

yangbolu1991 replied:
Will remove it.

hauke commented on target/linux/layerscape/armv7/config-4.14:
```diff
@@ -0,0 +1,925 @@
+CONFIG_MFD_TPS65217=y
+CONFIG_MFD_TPS65218=y
+CONFIG_MFD_TPS6586X=y
+CONFIG_MFD_TPS65910=y
```
Are all these MFD configuration options needed?

yangbolu1991 replied:
Will remove unused options.

hauke commented on target/linux/layerscape/armv7/config-4.14:
```diff
@@ -0,0 +1,925 @@
+# CONFIG_MTD_DATAFLASH_WRITE_VERIFY is not set
+CONFIG_MTD_M25P80=y
+CONFIG_MTD_NAND=y
+CONFIG_MTD_NAND_BRCMNAND=y
```
Do you have a broadcom NAND controller in the chip?

yangbolu1991 replied:
Will remove it.

hauke commented on target/linux/layerscape/armv7/config-4.14:
```diff
@@ -0,0 +1,925 @@
+# CONFIG_NET_VENDOR_TI is not set
+# CONFIG_NET_VENDOR_VIA is not set
+# CONFIG_NET_VENDOR_WIZNET is not set
+# CONFIG_NEW_LEDS is not set
```
Please do not deactivate all these CONFIG_NET_VENDOR options, the generic config will activate them and takes care of the individual drivers. CONFIG_NEW_LEDS should also be activated

biwenli replied:
CONFIG_NET_VENDOR_FREESCALE  is automatically activated when ARCH_LAYERSCAPE is activated

yangbolu1991 replied:
Will enable them in default. These may be used.

hauke commented on target/linux/layerscape/armv7/config-4.14:
```diff
@@ -0,0 +1,925 @@
+CONFIG_TRACING_EVENTS_GPIO=y
+CONFIG_TREE_RCU=y
+CONFIG_TREE_SRCU=y
+CONFIG_TUN=y
```
This should be activated by kmod-tun and not here.

yangbolu1991 replied:
Will remove it.

hauke commented on target/linux/layerscape/armv7/config-4.14:
```diff
@@ -0,0 +1,925 @@
+CONFIG_UTS_NS=y
+CONFIG_VDSO=y
+CONFIG_VECTORS_BASE=0xffff0000
+CONFIG_VETH=y
```
Please do not build veth into the kernel

yangbolu1991 replied:
Will remove it.

hauke reviewed (commented):
I only commended on some of the config files, but the comments apply to all of them.
In general we prefer that the target specific kernel configuration is pretty small and all the extra network features and extra drivers are controlled in a central manner with the kernel configuration in target/linux/config-4.* and the packages.
Feel free to propose changes there.
Please squash the commit " layerscape: refresh patches-4.14 " into the first one adding these patches

yangbolu1991 commented:
Changes for v2:
- Addressed comments on config-4.14 files for each arch.
- squashed the commit " layerscape: refresh patches-4.14 " into the first patch.

xback commented on target/linux/layerscape/patches-4.14/201-config-support-layerscape.patch:
```diff
@@ -0,0 +1,302 @@
+Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
+Signed-off-by: Horia Geantă <horia.geanta@nxp.com>
+Signed-off-by: Horia Geantă horia.geanta@nxp.com
+Signed-off-by: Ioana Ciornei <ioana.ciornei@nxp.com>
```
Double signed-off line here, with 1 having an invalid format

xback commented on target/linux/layerscape/patches-4.14/301-arch-support-layerscape.patch:
```diff
@@ -0,0 +1,327 @@
+Signed-off-by: Haiying Wang <Haiying.wang@freescale.com>
+Signed-off-by: Haiying Wang <Haiying.Wang@freescale.com>
+Signed-off-by: Horia Geantă <horia.geanta@nxp.com>
+Signed-off-by: Jerry Huang <Chang-Ming.Huang@freescale.com>
```
same here

biwenli replied:
Will remove it

xback commented on target/linux/layerscape/patches-4.14/301-arch-support-layerscape.patch:
```diff
@@ -0,0 +1,327 @@
+Signed-off-by: Zhao Qiang <B45475@freescale.com>
+Signed-off-by: Zhao Qiang <qiang.zhao@nxp.com>
+Signed-off-by: Biwen Li <biwen.li@nxp.com>
+---
```
and here

biwenli replied:
Will remove it

yangbolu1991 commented:
@mcbridematt We still use SDK DPAA/FMan driver in default. I think it should be not problem for Traverse.
Thanks a lot.

yangbolu1991 commented:
@xback Thanks a lot for your comments.
I have update the pull request for v3 to remove the duplicate Signed-off-by.

Regarding to the invalidate format. That's given by the author and linux community also accepted such format for name. That's not problem I think.
Signed-off-by: Horia Geantă <horia.geanta@nxp.com>

lucize commented:
sidenote, will ls1042 be supported ?

yangbolu1991 commented:
@lucize I think the ls1042 you said should be some variant of LS product, even I didn't find it in nxp.com.
https://www.nxp.com/products/processors-and-microcontrollers/arm-based-processors-and-mcus/qoriq-layerscape-arm-processors:QORIQ-ARM

We didn't plan to support variant of LS silicon. But it should be just core number difference.
The software could be just reused. If not work, only minor changes on dts are required I think.
Thanks a lot.

lucize commented:
I think is on the Cisco rv345, ok, hopefully I can find a way to flash it

mcbridematt commented:
@lucize The 1024 is actually the Mindspeed C2000 that NXP/Freescale acquired some years ago, target/layerscape will not work with it

yangbolu1991 commented:
Any comments?
Thanks a lot.

biwenli committed:
layerscape: drop kernel 4.9 support

This patch is to drop kernel 4.9 support.

Signed-off-by: Biwen Li <biwen.li@nxp.com>
Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>

yangbolu1991 commented:
Rebased the patches to fix conflicts.
Please help to review and merge.

Thanks a lot:)

ynezz commented:
@yangbolu1991 I don't have any of the devices, so I can't run time test it, but I've at least started built testing this PR with following config:

```
CONFIG_TARGET_layerscape=y
CONFIG_TARGET_layerscape_armv8_64b=y
CONFIG_TARGET_MULTI_PROFILE=y
CONFIG_TARGET_DEVICE_layerscape_armv8_64b_DEVICE_ls1012afrwy=y
CONFIG_TARGET_DEVICE_layerscape_armv8_64b_DEVICE_ls1012ardb=y
CONFIG_TARGET_DEVICE_layerscape_armv8_64b_DEVICE_ls1043ardb=y
CONFIG_TARGET_DEVICE_layerscape_armv8_64b_DEVICE_ls1043ardb-sdboot=y
CONFIG_TARGET_DEVICE_layerscape_armv8_64b_DEVICE_ls1046ardb=y
CONFIG_TARGET_DEVICE_layerscape_armv8_64b_DEVICE_ls1046ardb-sdboot=y
CONFIG_TARGET_DEVICE_layerscape_armv8_64b_DEVICE_ls1088ardb=y
CONFIG_TARGET_DEVICE_layerscape_armv8_64b_DEVICE_ls1088ardb-sdboot=y
CONFIG_TARGET_DEVICE_layerscape_armv8_64b_DEVICE_ls2088ardb=y
CONFIG_TARGET_DEVICE_layerscape_armv8_64b_DEVICE_traverse-five64=y
CONFIG_ALL_KMODS=y
CONFIG_ALL_NONSHARED=y
CONFIG_TARGET_ALL_PROFILES=y
```

and it fails really soon with:

```
Applying /opt/devel/openwrt/openwrt.git/target/linux/layerscape/patches-4.14/807-usb-support-layerscape.patch using plaintext: 
patching file Documentation/devicetree/bindings/usb/dwc3.txt
patching file arch/arm64/include/asm/io.h
patching file drivers/usb/common/common.c
patching file drivers/usb/core/usb.c
patching file drivers/usb/dwc3/core.c
Hunk #11 succeeded at 1462 (offset 1 line).
patching file drivers/usb/dwc3/core.h
patching file drivers/usb/dwc3/ep0.c
patching file drivers/usb/dwc3/gadget.c
Hunk #1 succeeded at 3203 (offset -5 lines).
Hunk #2 succeeded at 3280 (offset -5 lines).
patching file drivers/usb/dwc3/host.c
patching file drivers/usb/gadget/udc/fsl_udc_core.c
patching file drivers/usb/gadget/udc/fsl_usb2_udc.h
patching file drivers/usb/host/Kconfig
patching file drivers/usb/host/ehci-fsl.c
patching file drivers/usb/host/ehci-fsl.h
patching file drivers/usb/host/ehci-hub.c
patching file drivers/usb/host/ehci.h
patching file drivers/usb/host/fsl-mph-dr-of.c
patching file drivers/usb/host/xhci-hub.c
patching file drivers/usb/host/xhci-plat.c
patching file drivers/usb/host/xhci-ring.c
Hunk #1 succeeded at 1969 (offset 33 lines).
Hunk #2 succeeded at 1997 (offset 33 lines).
patching file drivers/usb/host/xhci.c
patching file drivers/usb/host/xhci.h
Hunk #1 FAILED at 1794.
Hunk #2 FAILED at 1830.
2 out of 2 hunks FAILED -- saving rejects to file drivers/usb/host/xhci.h.rej
patching file drivers/usb/phy/phy-fsl-usb.c
patching file drivers/usb/phy/phy-fsl-usb.h
patching file include/linux/usb.h
patching file include/linux/usb/of.h
Patch failed!  Please fix /opt/devel/openwrt/openwrt.git/target/linux/layerscape/patches-4.14/807-usb-support-layerscape.patch!
make[4]: *** [/opt/devel/openwrt/openwrt.git/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-4.14.87/.prepared_80301eed0c78ed6947a3999fdea2d91a] Error 1
make[4]: Leaving directory `/opt/devel/openwrt/openwrt.git/target/linux/layerscape'
make[3]: *** [compile] Error 2
```

yangbolu1991 commented:
@ynezz Oh... Maybe OpenWrt code base upgrade kernel 4.14, this usb patch should be rebased.
Let me check.

yangbolu1991 commented:
@ynezz Updated the pull request. The build passed on my side.
Changes:
- Fixed conflicts when applied target/linux/layerscape/patches-4.14/807-usb-support-layerscape.patch to kernel 4.14.87

Thanks.

ynezz commented:
@yangbolu1991 FYI, I'm not your personal build tester, you should do your homework better :-) Is it that hard to do `make target/linux/{clean,compile} -j$(nproc)` and confirm, that it builds for you at least?

```
CONFIG_TARGET_layerscape=y
CONFIG_TARGET_layerscape_armv8_64b=y
CONFIG_TARGET_MULTI_PROFILE=y
CONFIG_TARGET_DEVICE_layerscape_armv8_64b_DEVICE_ls1012afrwy=y
CONFIG_TARGET_DEVICE_layerscape_armv8_64b_DEVICE_ls1012ardb=y
CONFIG_TARGET_DEVICE_layerscape_armv8_64b_DEVICE_ls1043ardb=y
CONFIG_TARGET_DEVICE_layerscape_armv8_64b_DEVICE_ls1043ardb-sdboot=y
CONFIG_TARGET_DEVICE_layerscape_armv8_64b_DEVICE_ls1046ardb=y
CONFIG_TARGET_DEVICE_layerscape_armv8_64b_DEVICE_ls1046ardb-sdboot=y
CONFIG_TARGET_DEVICE_layerscape_armv8_64b_DEVICE_ls1088ardb=y
CONFIG_TARGET_DEVICE_layerscape_armv8_64b_DEVICE_ls1088ardb-sdboot=y
CONFIG_TARGET_DEVICE_layerscape_armv8_64b_DEVICE_ls2088ardb=y
CONFIG_TARGET_DEVICE_layerscape_armv8_64b_DEVICE_traverse-five64=y
CONFIG_ALL_KMODS=y
CONFIG_ALL_NONSHARED=y
CONFIG_TARGET_ALL_PROFILES=y
```

leads to the following error now:

```
rm -f /opt/devel/openwrt/openwrt.git/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-4.14.87/localversion
/opt/devel/openwrt/openwrt.git/scripts/kconfig.pl  + /opt/devel/openwrt/openwrt.git/target/linux/generic/config-4.14 /opt/devel/openwrt/openwrt.git/target/linux/layerscape/armv8_64b/config-4.14 > /opt/devel/openwrt/openwrt.git/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-4.14.87/.config.target
awk '/^(#[[:space:]]+)?CONFIG_KERNEL/{sub("CONFIG_KERNEL_","CONFIG_");print}' /opt/devel/openwrt/openwrt.git/.config >> /opt/devel/openwrt/openwrt.git/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-4.14.87/.config.target
echo "# CONFIG_KALLSYMS_EXTRA_PASS is not set" >> /opt/devel/openwrt/openwrt.git/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-4.14.87/.config.target
echo "# CONFIG_KALLSYMS_ALL is not set" >> /opt/devel/openwrt/openwrt.git/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-4.14.87/.config.target
echo "CONFIG_KALLSYMS_UNCOMPRESSED=y" >> /opt/devel/openwrt/openwrt.git/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-4.14.87/.config.target
/opt/devel/openwrt/openwrt.git/scripts/package-metadata.pl kconfig /opt/devel/openwrt/openwrt.git/tmp/.packageinfo /opt/devel/openwrt/openwrt.git/.config 4.14 > /opt/devel/openwrt/openwrt.git/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-4.14.87/.config.override
/opt/devel/openwrt/openwrt.git/scripts/kconfig.pl 'm+' '+' /opt/devel/openwrt/openwrt.git/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-4.14.87/.config.target /dev/null /opt/devel/openwrt/openwrt.git/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-4.14.87/.config.override > /opt/devel/openwrt/openwrt.git/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-4.14.87/.config.set
mv /opt/devel/openwrt/openwrt.git/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-4.14.87/.config.set /opt/devel/openwrt/openwrt.git/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-4.14.87/.config.old
grep -v INITRAMFS /opt/devel/openwrt/openwrt.git/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-4.14.87/.config.old > /opt/devel/openwrt/openwrt.git/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-4.14.87/.config.set
echo 'CONFIG_INITRAMFS_SOURCE=""' >> /opt/devel/openwrt/openwrt.git/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-4.14.87/.config.set
echo '# CONFIG_INITRAMFS_FORCE is not set' >> /opt/devel/openwrt/openwrt.git/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-4.14.87/.config.set
rm -rf /opt/devel/openwrt/openwrt.git/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/modules
cmp -s /opt/devel/openwrt/openwrt.git/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-4.14.87/.config.set /opt/devel/openwrt/openwrt.git/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-4.14.87/.config.prev || { cp /opt/devel/openwrt/openwrt.git/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-4.14.87/.config.set /opt/devel/openwrt/openwrt.git/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-4.14.87/.config; cp /opt/devel/openwrt/openwrt.git/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-4.14.87/.config.set /opt/devel/openwrt/openwrt.git/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-4.14.87/.config.prev; }
export MAKEFLAGS= ; [ -d /opt/devel/openwrt/openwrt.git/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-4.14.87/user_headers ] || make -C /opt/devel/openwrt/openwrt.git/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-4.14.87 HOSTCFLAGS="-O2 -I/opt/devel/openwrt/openwrt.git/staging_dir/host/include  -Wall -Wmissing-prototypes -Wstrict-prototypes" CROSS_COMPILE="aarch64-openwrt-linux-musl-" ARCH="arm64" KBUILD_HAVE_NLS=no KBUILD_BUILD_USER="" KBUILD_BUILD_HOST="" KBUILD_BUILD_TIMESTAMP="Wed Dec 12 17:01:31 2018" KBUILD_BUILD_VERSION="0" HOST_LOADLIBES="-L/opt/devel/openwrt/openwrt.git/staging_dir/host/lib" CONFIG_SHELL="bash" V=''  cmd_syscalls= KERNELRELEASE=4.14.87 CC="aarch64-openwrt-linux-musl-gcc" INSTALL_HDR_PATH=/opt/devel/openwrt/openwrt.git/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-4.14.87/user_headers headers_install
grep '=[ym]' /opt/devel/openwrt/openwrt.git/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-4.14.87/.config.set | LC_ALL=C sort | mkhash md5 > /opt/devel/openwrt/openwrt.git/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-4.14.87/.vermagic
touch /opt/devel/openwrt/openwrt.git/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-4.14.87/.configured
rm -f /opt/devel/openwrt/openwrt.git/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-4.14.87/vmlinux /opt/devel/openwrt/openwrt.git/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-4.14.87/System.map
make -C /opt/devel/openwrt/openwrt.git/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-4.14.87 HOSTCFLAGS="-O2 -I/opt/devel/openwrt/openwrt.git/staging_dir/host/include  -Wall -Wmissing-prototypes -Wstrict-prototypes" CROSS_COMPILE="aarch64-openwrt-linux-musl-" ARCH="arm64" KBUILD_HAVE_NLS=no KBUILD_BUILD_USER="" KBUILD_BUILD_HOST="" KBUILD_BUILD_TIMESTAMP="Wed Dec 12 17:01:31 2018" KBUILD_BUILD_VERSION="0" HOST_LOADLIBES="-L/opt/devel/openwrt/openwrt.git/staging_dir/host/lib" CONFIG_SHELL="bash" V=''  cmd_syscalls= KERNELRELEASE=4.14.87 CC="aarch64-openwrt-linux-musl-gcc" modules
make[5]: Entering directory `/opt/devel/openwrt/openwrt.git/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-4.14.87'
scripts/kconfig/conf  --silentoldconfig Kconfig
net/sched/Kconfig:44: warning: menuconfig statement without prompt
.config:1335:warning: override: DEFAULT_DEADLINE changes choice state
.config:5666:warning: override: SPARSEMEM_MANUAL changes choice state
*
* Restart config...
*
*
* Serial ATA and Parallel ATA drivers (libata)
*
Serial ATA and Parallel ATA drivers (libata) (ATA) [M/n/y/?] m
  Verbose ATA error reporting (ATA_VERBOSE_ERROR) [N/y/?] n
  SATA Port Multiplier support (SATA_PMP) [N/y/?] n
  *
  * Controllers with non-SFF native interface
  *
  AHCI SATA support (SATA_AHCI) [M/n/?] m
  Platform AHCI SATA support (SATA_AHCI_PLATFORM) [N/m/?] n
  CEVA AHCI SATA support (AHCI_CEVA) [N/m/?] n
  APM X-Gene 6.0Gbps AHCI SATA host controller support (AHCI_XGENE) [N/m/?] (NEW) aborted!

Console input/output is redirected. Run 'make oldconfig' to update configuration.

make[7]: *** [silentoldconfig] Error 1
make[6]: *** [silentoldconfig] Error 2

The present kernel configuration has modules disabled.
Type 'make config' and enable loadable module support.
Then build a kernel with module support enabled.

make[5]: *** [modules] Error 1
make[5]: Leaving directory `/opt/devel/openwrt/openwrt.git/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/linux-4.14.87'
```

ynezz commented:
@yangbolu1991 And following config:

```
CONFIG_TARGET_layerscape=y
CONFIG_TARGET_layerscape_armv8_32b=y
CONFIG_TARGET_MULTI_PROFILE=y
CONFIG_TARGET_DEVICE_layerscape_armv8_32b_DEVICE_ls1012afrwy=y
CONFIG_TARGET_DEVICE_layerscape_armv8_32b_DEVICE_ls1012ardb=y
CONFIG_TARGET_DEVICE_layerscape_armv8_32b_DEVICE_ls1043ardb=y
CONFIG_TARGET_DEVICE_layerscape_armv8_32b_DEVICE_ls1046ardb=y
CONFIG_ALL_KMODS=y
CONFIG_ALL_NONSHARED=y
CONFIG_TARGET_ALL_PROFILES=y
```
leads to the following build error:

```
make[5]: Entering directory '/home/petr/testing/openwrt/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/linux-layerscape_armv8_32b/linux-4.14.87'
  CHK     include/config/kernel.release
  CHK     include/generated/uapi/linux/version.h
  CHK     include/generated/utsrelease.h
  CHK     include/generated/bounds.h
  CHK     include/generated/timeconst.h
  CHK     include/generated/asm-offsets.h
  CALL    scripts/checksyscalls.sh
  CHK     scripts/mod/devicetable-offsets.h
  CC [M]  drivers/usb/host/ehci-fsl.o
drivers/usb/host/ehci-fsl.c:42:10: fatal error: asm/fsl_pm.h: No such file or directory
 #include <asm/fsl_pm.h>
          ^~~~~~~~~~~~~~
compilation terminated.
scripts/Makefile.build:332: recipe for target 'drivers/usb/host/ehci-fsl.o' failed
make[8]: *** [drivers/usb/host/ehci-fsl.o] Error 1
```

ynezz commented:
@yangbolu1991 

```
CONFIG_TARGET_layerscape=y
CONFIG_TARGET_layerscape_armv7=y
CONFIG_TARGET_MULTI_PROFILE=y
CONFIG_TARGET_DEVICE_layerscape_armv7_DEVICE_ls1021aiot-sdboot=y
CONFIG_TARGET_DEVICE_layerscape_armv7_DEVICE_ls1021atwr=y
CONFIG_TARGET_DEVICE_layerscape_armv7_DEVICE_ls1021atwr-sdboot=y
CONFIG_ALL_KMODS=y
CONFIG_ALL_NONSHARED=y
CONFIG_TARGET_ALL_PROFILES=y
```
```
In file included from ./include/linux/kernel.h:11:0,
                 from ./include/linux/clk.h:16,
                 from drivers/usb/host/xhci-plat.c:14:
drivers/usb/host/xhci-plat.c: In function 'xhci_plat_probe':
./include/linux/bitops.h:7:24: warning: left shift count >= width of type [-Wshift-count-overflow]
 #define BIT(nr)   (1UL << (nr))
                        ^
drivers/usb/host/xhci.h:1845:31: note: in expansion of macro 'BIT'
 #define XHCI_DIS_U1U2_WHEN_U3 BIT(35)
                               ^~~
drivers/usb/host/xhci-plat.c:270:20: note: in expansion of macro 'XHCI_DIS_U1U2_WHEN_U3'
    xhci->quirks |= XHCI_DIS_U1U2_WHEN_U3;
                    ^~~~~~~~~~~~~~~~~~~~~
./include/linux/bitops.h:7:24: warning: left shift count >= width of type [-Wshift-count-overflow]
 #define BIT(nr)   (1UL << (nr))
                        ^
drivers/usb/host/xhci.h:1833:33: note: in expansion of macro 'BIT'
 #define XHCI_REVERSE_IN_OUT     BIT(32)
                                 ^~~
drivers/usb/host/xhci-plat.c:274:19: note: in expansion of macro 'XHCI_REVERSE_IN_OUT'
   xhci->quirks |= XHCI_REVERSE_IN_OUT;
                   ^~~~~~~~~~~~~~~~~~~
./include/linux/bitops.h:7:24: warning: left shift count >= width of type [-Wshift-count-overflow]
 #define BIT(nr)   (1UL << (nr))
                        ^
drivers/usb/host/xhci.h:1834:39: note: in expansion of macro 'BIT'
 #define XHCI_STOP_TRANSFER_IN_BLOCK   BIT(33)
                                       ^~~
drivers/usb/host/xhci-plat.c:278:19: note: in expansion of macro 'XHCI_STOP_TRANSFER_IN_BLOCK'
   xhci->quirks |= XHCI_STOP_TRANSFER_IN_BLOCK;
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~
./include/linux/bitops.h:7:24: warning: left shift count >= width of type [-Wshift-count-overflow]
 #define BIT(nr)   (1UL << (nr))
                        ^
drivers/usb/host/xhci.h:1835:32: note: in expansion of macro 'BIT'
 #define XHCI_STOP_EP_IN_U1     BIT(34)
                                ^~~
drivers/usb/host/xhci-plat.c:281:19: note: in expansion of macro 'XHCI_STOP_EP_IN_U1'
   xhci->quirks |= XHCI_STOP_EP_IN_U1;
                   ^~~~~~~~~~~~~~~~~~
drivers/usb/host/ehci-fsl.c:42:10: fatal error: asm/fsl_pm.h: No such file or directory
 #include <asm/fsl_pm.h>
          ^~~~~~~~~~~~~~
compilation terminated.
scripts/Makefile.build:332: recipe for target 'drivers/usb/host/ehci-fsl.o' failed
```

yangbolu1991 commented:
@ynezz 
I'm sorry for the trouble. I indeed missed to do build test with all kmods enabled, after rebasing patches to send out the original patch-set.
I have found several build issues in config and kernel patches after enabling all kmods. Fixed them, and built successfully now.

Changes:
- Fixed config-4.14 file for armv8_64b.
- Updated below kernel patches to fix build issues.
   202-core-linux-support-layerscape.patch
   706-dpaa2-virtualbridge-support-layerscape.patch
   807-usb-support-layerscape.patch

ynezz commented:
@yangbolu1991  Seems to be ok now, thanks.

I've just successfully build tested following configs:

```
981067a layerscape: use MKUBIFS_OPTS for per-device ubi parameters
520e45a layerscape: drop kernel 4.9 support
c15872a layerscape: add LS1021AIOT board support
87d5a33 layerscape: upgrade kernel to 4.14
bcb8592 kmod-dma-buf: fix build with external kernel
3198ec5 ipq806x: fix and cleanup dts for NEC WG2600HP
```

```
CONFIG_TARGET_layerscape=y
CONFIG_TARGET_layerscape_armv8_64b=y
CONFIG_TARGET_MULTI_PROFILE=y
CONFIG_TARGET_DEVICE_layerscape_armv8_64b_DEVICE_ls1012afrwy=y
CONFIG_TARGET_DEVICE_layerscape_armv8_64b_DEVICE_ls1012ardb=y
CONFIG_TARGET_DEVICE_layerscape_armv8_64b_DEVICE_ls1043ardb=y
CONFIG_TARGET_DEVICE_layerscape_armv8_64b_DEVICE_ls1043ardb-sdboot=y
CONFIG_TARGET_DEVICE_layerscape_armv8_64b_DEVICE_ls1046ardb=y
CONFIG_TARGET_DEVICE_layerscape_armv8_64b_DEVICE_ls1046ardb-sdboot=y
CONFIG_TARGET_DEVICE_layerscape_armv8_64b_DEVICE_ls1088ardb=y
CONFIG_TARGET_DEVICE_layerscape_armv8_64b_DEVICE_ls1088ardb-sdboot=y
CONFIG_TARGET_DEVICE_layerscape_armv8_64b_DEVICE_ls2088ardb=y
CONFIG_TARGET_DEVICE_layerscape_armv8_64b_DEVICE_traverse-five64=y
CONFIG_ALL_KMODS=y
CONFIG_ALL_NONSHARED=y
CONFIG_TARGET_ALL_PROFILES=y
```

```
CONFIG_TARGET_layerscape=y
CONFIG_TARGET_layerscape_armv8_32b=y
CONFIG_TARGET_MULTI_PROFILE=y
CONFIG_TARGET_DEVICE_layerscape_armv8_32b_DEVICE_ls1012afrwy=y
CONFIG_TARGET_DEVICE_layerscape_armv8_32b_DEVICE_ls1012ardb=y
CONFIG_TARGET_DEVICE_layerscape_armv8_32b_DEVICE_ls1043ardb=y
CONFIG_TARGET_DEVICE_layerscape_armv8_32b_DEVICE_ls1046ardb=y
CONFIG_ALL_KMODS=y
CONFIG_ALL_NONSHARED=y
CONFIG_TARGET_ALL_PROFILES=y
```

```
CONFIG_TARGET_layerscape=y
CONFIG_TARGET_layerscape_armv7=y
CONFIG_TARGET_MULTI_PROFILE=y
CONFIG_TARGET_DEVICE_layerscape_armv7_DEVICE_ls1021aiot-sdboot=y
CONFIG_TARGET_DEVICE_layerscape_armv7_DEVICE_ls1021atwr=y
CONFIG_TARGET_DEVICE_layerscape_armv7_DEVICE_ls1021atwr-sdboot=y
CONFIG_ALL_KMODS=y
CONFIG_ALL_NONSHARED=y
CONFIG_TARGET_ALL_PROFILES=y
```

ynezz commented:
@mcbridematt Would be nice if you could run test this on `traverse-five64` your board.

mcbridematt commented:
@yangbolu1991 It looks like the usb patch needs to be refreshed, when the pull request is rebased onto the master branch it fails:
```
Applying /home/matt/openwrt-ls1043-upstream/target/linux/layerscape/patches-4.14/807-usb-support-layerscape.patch using plaintext:
patching file Documentation/devicetree/bindings/usb/dwc3.txt
patching file arch/arm64/include/asm/io.h
patching file drivers/usb/common/common.c
patching file drivers/usb/core/usb.c
patching file drivers/usb/dwc3/core.c
patching file drivers/usb/dwc3/core.h
patching file drivers/usb/dwc3/ep0.c
patching file drivers/usb/dwc3/gadget.c
patching file drivers/usb/dwc3/host.c
patching file drivers/usb/gadget/udc/fsl_udc_core.c
patching file drivers/usb/gadget/udc/fsl_usb2_udc.h
patching file drivers/usb/host/Kconfig
patching file drivers/usb/host/ehci-fsl.c
patching file drivers/usb/host/ehci-fsl.h
patching file drivers/usb/host/ehci-hub.c
patching file drivers/usb/host/ehci.h
patching file drivers/usb/host/fsl-mph-dr-of.c
patching file drivers/usb/host/xhci-hub.c
patching file drivers/usb/host/xhci-plat.c
patching file drivers/usb/host/xhci-ring.c
patching file drivers/usb/host/xhci.c
Hunk #1 succeeded at 1599 (offset 18 lines).
Hunk #2 succeeded at 5015 (offset 34 lines).
patching file drivers/usb/host/xhci.h
Hunk #3 FAILED at 1842.
1 out of 3 hunks FAILED -- saving rejects to file drivers/usb/host/xhci.h.rej
patching file drivers/usb/phy/phy-fsl-usb.c
patching file drivers/usb/phy/phy-fsl-usb.h
patching file include/linux/usb.h
patching file include/linux/usb/of.h
```

yangbolu1991 commented:
@mcbridematt I saw kernel 4.14 in openwrt has upgraded 4.14.88.
Let me check and update the pull request soon.
Thanks.

mcbridematt commented:
mac80211 (in backports/compat-wireless) also does not build for 64b, I am trying to work out why:
```
mkdir -p /home/matt/openwrt-ls1043-upstream/bin/targets/layerscape/armv8_64b/packages /home/matt/openwrt-ls1043-upstream/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/backports-4.19-rc5-1/ipkg-aarch64_generic/kmod-mac80211/CONTROL /home/matt/openwrt-ls1043-upstream/staging_dir/target-aarch64_generic_musl/pkginfo
ERROR: module '/home/matt/openwrt-ls1043-upstream/build_dir/target-aarch64_generic_musl/linux-layerscape_armv8_64b/backports-4.19-rc5-1/net/mac80211/mac80211.ko' is missing.
Makefile:568: recipe for target '/home/matt/openwrt-ls1043-upstream/bin/targets/layerscape/armv8_64b/packages/kmod-mac80211_4.14.87+4.19-rc5-1-1_aarch64_generic.ipk' failed
make[3]: *** [/home/matt/openwrt-ls1043-upstream/bin/targets/layerscape/armv8_64b/packages/kmod-mac80211_4.14.87+4.19-rc5-1-1_aarch64_generic.ipk] Error 1
make[3]: Leaving directory '/home/matt/openwrt-ls1043-upstream/package/kernel/mac80211'
```

mcbridematt commented:
@yangbolu1991 mac80211 does not compile because the layerscape config unsets CONFIG_CRYPTO_ARC4.

Please remove this line from each config-4.14 file:
```
# CONFIG_CRYPTO_ARC4 is not set
```

yangbolu1991 commented:
@mcbridematt updated the pull request. Actually I didn't see build issue with mac80211 but I did some cleanup for crypto configs anyway. 
Changes:
- Rebased kernel 4.14 usb patch on linux 4.14.88.
- Cleaned up crypto configs in config-4.14.

Would you please help to verify traverse-five64 with latest pull request?
Let's get bugs fixed (if have) and get pull request merged soon.
Thanks in advance :)

ynezz commented:
@yangbolu1991 I'm really starting to wonder, if you're compile testing all your changes on all layerscape targets, because it fails for me again:

```
0d23fd2 treewide: dts: Remove default-state=off property from all gpio LED nodes
1e4ee63 ath79: add support for TP-Link Archer A7
d9753f0 ath79, brcm63xx: Fix debounce-interval properties in gpio-keys-polled nodes
7bd2595 ath79: ag71xx: Fix code formatting in ag71xx_probe
a29df78 ath79: ag71xx: Unify debug output with ar71xx and make debugging easier
22f9afa ar71xx: ag71xx: Replace duplicate debugging code with simple function call
d2a3b9a layerscape: use MKUBIFS_OPTS for per-device ubi parameters
dfb6cf0 layerscape: drop kernel 4.9 support
994af32 layerscape: add LS1021AIOT board support
b230b3d layerscape: upgrade kernel to 4.14
```

```
CONFIG_TARGET_layerscape=y
CONFIG_TARGET_layerscape_armv8_32b=y
CONFIG_TARGET_MULTI_PROFILE=y
CONFIG_TARGET_DEVICE_layerscape_armv8_32b_DEVICE_ls1012afrwy=y
CONFIG_TARGET_DEVICE_layerscape_armv8_32b_DEVICE_ls1012ardb=y
CONFIG_TARGET_DEVICE_layerscape_armv8_32b_DEVICE_ls1043ardb=y
CONFIG_TARGET_DEVICE_layerscape_armv8_32b_DEVICE_ls1046ardb=y
CONFIG_ALL_KMODS=y
CONFIG_ALL_NONSHARED=y
CONFIG_TARGET_ALL_PROFILES=y
```

```
In file included from ./include/linux/device.h:16:0,
                 from ./include/linux/of_platform.h:14,
                 from drivers/net/ethernet/freescale/sdk_dpaa/mac-api.c:44:
./include/linux/ioport.h:26:1: internal compiler error: Segmentation fault
 };
 ^
0x9e090f crash_signal
        /home/petr/testing/openwrt/build_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-7.4.0_musl_eabi/gcc-7.4.0/gcc/toplev.c:337
0x102f528 _cpp_process_line_notes
        /home/petr/testing/openwrt/build_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-7.4.0_musl_eabi/gcc-7.4.0/libcpp/lex.c:1165
0x102f650 _cpp_skip_block_comment
        /home/petr/testing/openwrt/build_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-7.4.0_musl_eabi/gcc-7.4.0/libcpp/lex.c:1224
0x103255c _cpp_lex_direct
        /home/petr/testing/openwrt/build_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-7.4.0_musl_eabi/gcc-7.4.0/libcpp/lex.c:2818
0x1036134 lex_expansion_token
        /home/petr/testing/openwrt/build_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-7.4.0_musl_eabi/gcc-7.4.0/libcpp/macro.c:3029
0x1039540 create_iso_definition
        /home/petr/testing/openwrt/build_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-7.4.0_musl_eabi/gcc-7.4.0/libcpp/macro.c:3201
0x1039540 _cpp_create_definition
        /home/petr/testing/openwrt/build_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-7.4.0_musl_eabi/gcc-7.4.0/libcpp/macro.c:3277
0x1023801 do_define
        /home/petr/testing/openwrt/build_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-7.4.0_musl_eabi/gcc-7.4.0/libcpp/directives.c:644
0x1024fcf _cpp_handle_directive
        /home/petr/testing/openwrt/build_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-7.4.0_musl_eabi/gcc-7.4.0/libcpp/directives.c:547
0x1032f2c _cpp_lex_token
        /home/petr/testing/openwrt/build_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-7.4.0_musl_eabi/gcc-7.4.0/libcpp/lex.c:2566
0x1038ecf cpp_get_token_1
        /home/petr/testing/openwrt/build_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-7.4.0_musl_eabi/gcc-7.4.0/libcpp/macro.c:2523
0x618c8e c_lex_with_flags(tree_node**, unsigned int*, unsigned char*, int)
        /home/petr/testing/openwrt/build_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-7.4.0_musl_eabi/gcc-7.4.0/gcc/c-family/c-lex.c:398
0x5b5410 c_lex_one_token
        /home/petr/testing/openwrt/build_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-7.4.0_musl_eabi/gcc-7.4.0/gcc/c/c-parser.c:251
0x5d95c2 c_parser_peek_token  
        /home/petr/testing/openwrt/build_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-7.4.0_musl_eabi/gcc-7.4.0/gcc/c/c-parser.c:435
0x5d95c2 c_parser_next_token_is
        /home/petr/testing/openwrt/build_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-7.4.0_musl_eabi/gcc-7.4.0/gcc/c/c-parser.h:161
0x5d95c2 c_parser_next_token_is_not
        /home/petr/testing/openwrt/build_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-7.4.0_musl_eabi/gcc-7.4.0/gcc/c/c-parser.h:170
0x5d95c2 c_parser_translation_unit
        /home/petr/testing/openwrt/build_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-7.4.0_musl_eabi/gcc-7.4.0/gcc/c/c-parser.c:1346
0x5d95c2 c_parse_file()
        /home/petr/testing/openwrt/build_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-7.4.0_musl_eabi/gcc-7.4.0/gcc/c/c-parser.c:18183
0x61e992 c_common_parse_file()
        /home/petr/testing/openwrt/build_dir/toolchain-arm_cortex-a15+neon-vfpv4_gcc-7.4.0_musl_eabi/gcc-7.4.0/gcc/c-family/c-opts.c:1161
Please submit a full bug report,
with preprocessed source if appropriate.
Please include the complete backtrace with any bug report.
See <http://www.lede-project.org/bugs/> for instructions.
make[10]: *** [drivers/net/ethernet/freescale/sdk_dpaa/mac-api.o] Error 1
```

mcbridematt commented:
Tested working (incl. mac80211/ath9k/ath10k) on the Traverse board.

yangbolu1991 commented:
@mcbridematt 
Good to know Traverse board worked. Thank you.

@ynezz 
Regarding to restool package, it's "Layerscape DPAA2 dynamical management tool".
It's only for ls1088ardb/ls2088ardb which were DPAA2 platforms and only used with 64-bit.
I will send a patch to fix the DEPENDS to make it built only for ls1088ardb/ls2088ardb.

Regarding to the build issue, I just rebased the patches on my side and double checked.
I didn't see build issue. Attached build log. Here are my steps in log.
I will re-clone source and build my patches to see whether I can reproduce.

make target/linux/clean
make -j16
./scripts/diffconfig.sh > layerscape-armv8_32b.config
cat layerscape-armv8_32b.config
git log --oneline

yangbolu1991 commented:
@ynezz 
Re-cloned source code and did build test. I didn't find build issue. Attached build log.
[build-pass.txt](https://github.com/openwrt/openwrt/files/2688831/build-pass.txt)

That's strange because I only updated usb patch since build succeeded on your side last time.
Your build error seemed to be from compiler.
Please let me know your steps if you are still facing the issue.

Thanks a lot:)

ynezz commented:
@yangbolu1991 

I've just re-run `layerscape_armv8_32b` build from scratch (running `git clean -xfd`), and I didn't hit the gcc segfault error this time (I quite expected this), but I'm seeing two other issues which might be probably related.

I'm running all my builds in verbose mode (`make V=s`) to see warnings etc. and in case of `layerscape_armv8_32b` build for 4.19 it is leading to kernel config questions when the kernel is about to be configured, which probably means, that the kernel config for this target isn't probably in proper condition (or there might be other issue as well), so maybe some config symbols/modules are not enabled as needed:

```
scripts/kconfig/conf  --syncconfig Kconfig
net/sched/Kconfig:44: warning: menuconfig statement without prompt
.config:3711:warning: symbol value 'm' invalid for NF_NAT_REDIRECT
*
* Restart config...
*
*
* Multiple platform selection
*
*
* CPU Core family selection
*
ARMv4 based platforms (FA526) (ARCH_MULTI_V4) [N/y/?] (NEW)
```

Which in turn probably leads to the following build failure?

```
ERROR: module 'target-arm_cortex-a15+neon-vfpv4_musl_eabi/linux-layerscape_armv8_32b/linux-4.19.9/drivers/net/phy/mdio-gpio.ko' is missing.
modules/netdevices.mk:130: recipe for target 'layerscape/armv8_32b/packages/kmod-mdio-gpio_4.19.9-1_arm_cortex-a15_neon-vfpv4.ipk' failed
```

mcbridematt commented:
Hi @yangbolu1991 
The layerscape build is broken since the 4.14 version was updated to 4.14.99:
drivers/vfio/vfio_iommu_type1.c: In function 'vfio_iommu_type1_attach_group':
drivers/vfio/vfio_iommu_type1.c:1353:7: warning: 'resv_msi_base' may be used uninitialized in this function [-Wmaybe-uninitialized]
   ret = iommu_get_msi_cookie(domain->domain, resv_msi_base);
   ~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  CC      drivers/vfio/fsl-mc/vfio_fsl_mc.o
In file included from ./include/linux/scatterlist.h:9:0,
                 from ./include/linux/iommu.h:22,
                 from drivers/vfio/fsl-mc/vfio_fsl_mc.c:14:
drivers/vfio/fsl-mc/vfio_fsl_mc.c: In function 'vfio_fsl_mc_dprc_wait_for_response':
./arch/arm64/include/asm/io.h:122:45: error: expected expression before ')' token
        : "=r" (tmp) : "r" ((unsigned long)(v))  \
                                             ^
drivers/vfio/fsl-mc/vfio_fsl_mc.c:334:3: note: in expansion of macro '__iormb'
   __iormb();
   ^~~~~~~
./arch/arm64/include/asm/io.h:122:45: error: expected expression before ')' token
        : "=r" (tmp) : "r" ((unsigned long)(v))  \
                                             ^
drivers/vfio/fsl-mc/vfio_fsl_mc.c:336:3: note: in expansion of macro '__iormb'
   __iormb();
   ^~~~~~~

I think it is related to this commit which changed the semantics of iormb:
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v4.14.99&id=6636dc5e01c613f032c9140b4bd113e783c24546

Is the 4.19 patchset far away? There has been discussion on the mailing list about branching for OpenWrt 19.03 so we might need to fix this now.

yangbolu1991 commented:
Hi @mcbridematt ,
Thank you for pointing out this build issue.
Let me ask our VFIO IP owner to generate a fix-up patch quickly. Will let you know once I send out the pull request.

Regarding to 4.19, NXP will have a LSDK-19.03 release with kernel 4.19 support. After that, we can start porting work to openwrt kernel 4.19.
Thanks a lot.

yangbolu1991 commented:
@mcbridematt @hauke 
Please see fix-up patch here. Thanks!
https://github.com/openwrt/openwrt/pull/1875
